authors/apps/authentication/views.py

```diff
@@ -24,9 +31,68 @@ def post(self, request):
         # your own work later on. Get familiar with it.
         serializer = self.serializer_class(data=user)
         serializer.is_valid(raise_exception=True)
+
+        email = serializer.validated_data.get('email', None)
+        username = serializer.validated_data.get('username')
+
+        payload  = {
+            'email':email,
+            'username': username,
+            'exp': datetime.utcnow()
+                    + timedelta(minutes=60)
+        }
+        token = jwt.encode(payload, settings.SECRET_KEY, algorithm='HS256').decode('utf-8')
+        sender = os.getenv('EMAIL_HOST_USER')
+        site_link = get_current_site(request)
+        verification_link = 'http://' + site_link.domain + '/api/v1/users/verify/{}'.format(token)
+
+        email_subject = "Author's Haven Email verification"
+        message = render_to_string('verification_template.html', {
+            'title': email_subject,
+            'username': username,
+            'verification_link': verification_link
+        })
+
+        send_mail(email_subject, '', sender, [email, ], html_message=message)
+
         serializer.save()
+        message = {
+            "message": "User successfully created. Check email for verification link",
+            "user_info": serializer.data,
+            "token":token
+        }
 
-        return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(message, status=status.HTTP_201_CREATED)
+
+class VerifyAPIView(APIView):
+    
+    serializer_class = UserSerializer
+    def get(self, request, token):
+
+        try:
+            email = jwt.decode(token, settings.SECRET_KEY)['email']
+            username = jwt.decode(token, settings.SECRET_KEY)['username']
+            user = User.objects.get(email=email)
+
+            if user.is_verified:
+                message = {
+                    'message': 'Account already activated. Click on the link to continue',
+                    'login link': 'http://127.0.0.1:8000/api/v1/users/login'
```

Comment:
This won't always be localhost, e.g in the deployed version.

Reply:
Sure. I will edit that.